### DISCUSSION TITLE
easy way to change profile

### DISCUSSION THREAD

**Discussion by Yueh72** [sezanzeb/input-remapper#97] [Ideas]
Hi, i just have an idea to quickly change profile without open the application.

-> Add a system tray applet to switch between different profiles

I don't know if it's possible.

**Comment by sezanzeb** [comment]
I agree that it would be nice to have that. Should be possible, somebody just needs to do it

For example there is this project for pulseeffects that adds a taskbar tray: https://github.com/boomshop/pulseeffectstray

An application that is in the tray can control key-mapper via dbus with key-mapper-control: https://github.com/sezanzeb/key-mapper/blob/main/readme/usage.md#cli

**Comment by fvkd** [comment]
I hate to necromance an old post like this, but I've seen multiple posts where folks are asking for a way to toggle on/off presets for individual devices or switch presets to individual devices for various reasons. 
I can totally relate to this. My particular use case: I have a ThinkPad T14s Gen 1. A great machine. All except for the fact that a lot of these machines came out-of-the-box with one of the crappiest defects you could possibly deal with, dead keys. Like, dead as JFK. No one knows why or how and there's no solution except to flex your warranty if you have one. I have the top tier warranty package that includes on-site service. Myself and other folks I've ran into on Reddit and various forums know the local technicians that Lenovo sends to on-site service us, by first name at this point. The technicians know my Sphynx cat's name. Anyway, the dead keys affected are possibly the worst ones you could be without: Enter, backslash, backspace, F5, F9, F10, and some for some folks (mine works, praise the sun) the spacebar as well. 
This is an exectutive level laptop, with some wonderful security features. But it's a little difficult to get past the point of decrypting your partitions on boot without an enter key. I've totally effed up a dozen and a half keyboards by toting them along in my bookbag when I'm on the go and the only reason I had them with me was the enter key. I work in info sec, so I carry tools of my trade with me usually and some of them have been a major aid in this problem, like the digispark and my usb rubber ducks. That solution turns your personal attack plane from being a small table top to suddenly having your ass exposed, running backwards upwind in a corn field during a tornado, should your lose the digispark/rubber ducky or someone malacious steals it. A big ol' NOPE. The other solution: unencrypted drives, which are yuck, just ugh, yuck. yuck. 
Did I mention how much it sucks out loud to not have a functional enter key?
And while input-remapper isn't a solution on a level that I can decrypt my partitions and drives on boot, it sure does make me feel a little safer when I'm naked in a metaphoric corn field during a tornado. 
I'm resurrecting this post because I've read through the others on toggling input-remapper, and this one seems to be the closest to having a solution in the works for doing so without having to open the gui. 
Input-remapper has been a bit of a scythe for me in the metaphoric cornfield tornado. I've spent hours theory-crafting and streamlining my remappings for ergonomics and fingertravel. 
I've ended up borrowing a genious idea from https://github.com/Radvendii/trackpoint-clusters
Sacrificing my touchpad/trackpad buttons to be remapped as thumb-button clusters has made a world of difference, turning my onboard Thinkpad keyboard into an almost-as-feature-rich finger-weapon like the EZ-Ergodox keyboard when implemented with Input-Remapper. 
I run NixOS on all of my machines and share most of my dotfiles and setups between all of my hosts. I'm so spoiled on the ease of use from my remappings on the problematic Thinkpad that I have remapped most of my other keyboards and keys on my non-problematic machines to share most of the same remapping. 
Now, after much long, drawn out babble, I get to the point of this toggle post....
Sometimes having the touchpad/trackpad buttons remapped (left button remapped to 'Ctrl+Shift', middle touchpad button remapped to enter, and right touchpad button remapped to backspace) adheres to some things like web browsing. Some window managers don't play well with touchpad 'tap-to-click'. It would be just the bestest ever to be able to toggle an individual device's ipnut-remapper settings (am I the only person that has used about an hour totaled up of my life backspacing the type-o 'IPNUT' before I just made an alias or shell abbreviation for it? Having a dead backspace key being the reason for all this in the first place will lead to a prescription of Klonopin.) without the gui or tabbing over to the terminal. And so, I am going to make this happen for everyone. 
Over here on my side of this bastard keyboard, I'm nicknaming/calling this tray-sitting toggler "Project Tornado Scythe". 
As of now while typing my two years of frustrate out to yall, I just discovered Sezanzeb's reply here, and I see the terminal command for toggling when I follow the usage link. I've somehow overlooked that who-knows how many times reading here. I haven't made a macro remapping for it yet to see if it will be as simple as just that, remapping a key or two with a macro for that command or if a gui or tray-sitting toggler and some duct tape will be necessary. 
If it ends up being as simple as some remapping macros, well, I won't feel too bad for my necromancy as this solution is buried under a forgotten dungeon, which is buried under a graveyard, which is buried under a metaphoric corn field. 
Thank you Yueh72 for being OP, and most of all, so very many thank you's to Sezanzeb for the awesome tool that is Ipnut-Remapper. I reckon I'll get back here soon with some progress.
